### ISSUE TITLE
How to dedup subtring in one dataset?

### ISSUE THREAD

**Issue by lan2016286** [google-research/deduplicate-text-datasets#7]
For example, I have a dataset of many articles, is there any command to delete duplicate data in the above dataset?

**Comment by carlini** [comment]
Yeah, this can be done.

To start, you'll want to build a suffix array of the dataset (as described in the README). If you're using a new dataset that's not in TFDS, you'll need to merge all the examples together into a single file following the protocol in scripts/load_dataset.py.

From here, once you have a suffix array created, you can just run 

```cargo run selfsimilar_parallel [dataset_file]
cargo run collect_similar [dataset_file]```

And this will give you a collection of substrings that should be deleted from your dataset because they match other sequences.

**Comment by lan2016286** [comment]
In my dataset there are two passages as follows: "During his first year with Neuchatel Xamax FC, he did not play a single minute and on May 17, 2010 was loaned to FC Biel." and "During his first year for MU, he did not play a single minute and on 17 May 2010 was loaned to FC Biel."
After I ran "cargo run collect_similar [dataset_file]" found both duplicate substrings ", he did not play for a minute and on 17 May 2010 was loaned to FC Biel." in both paragraphs above. This I did not expect.
What I was expecting was to find a substring ", he did not play a minute and on 17 May 2010 was loaned to FC Biel." unique in the 2nd paragraph, this substring is duplicated with the substring in the previous paragraph.
